pkg/config/types/parse.go

```diff
@@ -80,7 +81,10 @@ func parseStringSlice(key string, values []string, typ reflect.Type) (any, error
 func parseDuration(key, value string) (string, error) {
 	duration, err := time.ParseDuration(value)
 	if err != nil {
-		return "", fmt.Errorf("parsing value: '%s' for key: '%s' to duration: %w", value, key, err)
+		// NB: this error is returned directly to the user, so we format it
+		return "", fmt.Errorf("config key: %q expects a valid duration value, received: %q.\n"+
+			"\tAccepted formats include combinations of 'h' (hours), 'm' (minutes), 's' (seconds), 'ms' (milliseconds), etc.\n"+
+			"\tExample: '2h45m', '30s', '100ms'", key, value)
```

Comment:
Please use `bacerrors.New(...)` that was introduced in #4536. It should look something similar to:
```
bacerrors.New("config key: %q expects a valid duration value, received: %q", key, value).
WithHint("Accepted formats ...").
WithCode(bacerrors.ValidationError)
WithComponent("Config")
```

The new error type simplifies error propagation and avoids unnecessary wrapping and we keep track of the root cause and its stacktrace.

The same applies for other errors. hint is optional but recommended